FilePersistence/src/version_control/GitRepository.cpp

```diff
@@ -318,30 +318,48 @@ CommitGraph GitRepository::commitGraph(QString startRevision, QString endRevisio
 
 const Commit* GitRepository::getCommit(QString revision) const
 {
-	Q_ASSERT(revision != WORKDIR);
 	Q_ASSERT(revision != INDEX);
+	if (revision == WORKDIR)
+	{
+		Commit* m = nullptr;
+		QDirIterator DirIter(workdirPath(), QDirIterator::Subdirectories);
+		while (DirIter.hasNext())
+		{
+			DirIter.next();
+			if (DirIter.fileInfo().isFile()){
+				QFile file(DirIter.filePath());
+				QByteArray arr = file.readAll();
+				char* content = arr.data();
+				std::unique_ptr<char[], CommitFileContentDeleter> content_ptr{arr.data(), [content](char*){free(content);}};
```

Comment:
In this case (unlike the code you copied from, that uses an external library - libgit2) you do not need to manually provide a deleter, and should definitely not call `free`. Simply remove the last argument of the constructor to `content_ptr`.